ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.cpp

```diff
@@ -57,7 +57,7 @@ static Tensor reduce_impl(
             rank);
     }
 
-    if (dim.size() == 1) {
+    if (dim.size() == 1 && rank == 4) {
```

Comment:
What happens if `dim.size() == 1` and rank is not 4? Do we need a fatal?

Reply:
Then the rest of the code runs, which is the previous functional behaviour.

Before, the code would go into `if (dim.size() == 1) {`
and then see that
`else if (dim[0] == 0)` is false becaues -2 != 0
Therefore that conditional is not executed and the rest of the code is called.
however, after an earlier fix to map negative indices to regular indices, -2 is updated to be 0
Therefore that conditional is executed, and it calls pad with a 2D input, which is not supported.
This change reverts the functionality to not execute the conditional and execute the rest of the code correctly.